File: src/main/java/sunshine/infrastructure/configuration/RestClientProperties.java
package sunshine.infrastructure.configuration;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "sunshine.client")
public record RestClientProperties(
    long connectTimeout,
    long responseTimeout,
    RetryProperties retry
) {
  public record RetryProperties(
      int maxAttempts,
      long backoff
  ) {
  }
}

File: src/main/kotlin/sunshine/infrastructure/configuration/RestClientProperties.kt
package sunshine.infrastructure.configuration

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "sunshine.client")
data class RestClientProperties(
    val connectTimeout: Long,
    val responseTimeout: Long,
    val retry: RetryProperties
) {
    data class RetryProperties(
        val maxAttempts: Int,
        val backoff: Long
    )
}
